src/actions/clusterActions.js

```diff
@@ -186,7 +186,10 @@ export function clusterLoadDetails(
 
       return cluster;
     } catch (error) {
-      if (error.status === StatusCodes.NotFound) {
+      if (
+        error.status === StatusCodes.NotFound ||
+        error.status === StatusCodes.Unathorized
```

Comment:
There's no need for the double check, because in the snippet you sent, the API returns a 404 response even if the user is unauthorized.
It is explained why in the comment there
```
// Returning 404 NOT FOUND on unauthorized request is common
// practice and aligns with v4 implementation as well.
```

Reply:
Yes, but check this...

> I noticed that the `error.status` when a cluster is not found is actually `401` whilst `error.response.status` is `404`. The browser identifies it as a `404` so I just assumed that we have to check this one instead.

This happens when a cluster is deleted and Happa tries to refresh the clusters list, when I found this in the API I couldn't help to think that this could be somehow related. Maye it is related with the auth token refresh instead, as Oliver is suggesting.

Also, removing a cluster when the user is not authorized and the cluster is in the store, makes sense. We should obviously change the message, but what happens after this check is valid for a `401 `, so what I am proposing is: let's merge this as an temporary fix (or temporary test) and see if we get less _x is undefined_ exceptions in the cluster list.

Reply:
BTW if someone wants to work on this, you can reproduce the error simply by creating a cluster, then deleting it and just leave the cluster list opened for a while.

Reply:
The server returns a `404` error, so the problem is on the client-side
![image](https://user-images.githubusercontent.com/13508038/76878298-eb378200-6874-11ea-8d6e-2a75163d5c10.png)

And IMO, the status code change happens in 2 places, and because of the structure of the response object, this  change can only happen in `/src/lib/giantswarmClientPatcher.js`, inside the catch call, where we change the status code to `401`

Reply:
Makes sense! 

Should we merge this so we can get rid of these errors? or you prefer to leave it like this for now?